src/core/net/dnssd_server.hpp

```diff
@@ -232,46 +236,111 @@ class Server : public InstanceLocator, private NonCopyable
-    void             HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
-    void             ProcessQuery(Message &aMessage, Message &aResponse, const Header &aRequestHeader);
-    Header::Response ResolveQuestion(const char *      aName,
-                                     const Question &  aQuestion,
-                                     Header &          aResponseHeader,
-                                     Message &         aResponseMessage,
-                                     uint8_t           aResolveKind,
-                                     NameCompressInfo &aCompressInfo);
-    static otError   AppendQuestion(const char *      aName,
-                                    const Question &  aQuestion,
-                                    Message &         aMessage,
-                                    NameCompressInfo &aCompressInfo);
-    static otError   AppendPtrRecord(Message &         aMessage,
-                                     const char *      aServiceName,
-                                     const char *      aInstanceName,
-                                     uint32_t          aTtl,
-                                     NameCompressInfo &aCompressInfo);
-    static otError   AppendSrvRecord(Message &         aMessage,
-                                     const char *      aInstanceName,
-                                     const char *      aHostName,
-                                     uint32_t          aTtl,
-                                     uint16_t          aPriority,
-                                     uint16_t          aWeight,
-                                     uint16_t          aPort,
-                                     NameCompressInfo &aCompressInfo);
-    static otError   AppendAaaaRecord(Message &           aMessage,
-                                      const char *        aHostName,
-                                      const Ip6::Address &aAddress,
-                                      uint32_t            aTtl,
-                                      NameCompressInfo &  aCompressInfo);
-    static otError   AppendServiceName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
-    static otError   AppendInstanceName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
-    static otError   AppendHostName(Message &aMessage, const char *aName, NameCompressInfo &aCompressInfo);
-    static void      IncResourceRecordCount(Header &aHeader, bool aAdditional);
-    static otError   FindNameComponents(const char *aName, const char *aDomain, NameComponentsOffsetInfo &aInfo);
-    static otError   FindPreviousLabel(const char *aName, uint8_t &aStart, uint8_t &aStop);
+#if OPENTHREAD_CONFIG_DNSSD_DISCOVERY_PROXY_ENABLE
+    enum : uint32_t
+    {
+        kDiscoveryProxyTimeout = OT_DNSSD_DISCOVERY_PROXY_TIMEOUT,
+    };
+
+    class DiscoveryProxyRequest
```

Comment:
```suggestion
    class DnsQueryTransaction
```

This sturct doesn't seem to be a Discovery Proxy request.

Reply:
Adding `DnsQueryTransaction` SGTM. But it would cause significant refactoring. 

I already have future plan to refactor `resolving by SRP` part of code. So, I would prefer to do refactoring later after we deliver the working features first. 

Thoughts?

Reply:
Let's postpone this major refactoring and deliver the working feature first.